### PR TITLE
Support team access

### PR DESCRIPTION
**Pull request by ryanwholey**
For team access I think it will be a nicer experience for the user to use the team name instead of the team ID. This should be okay because each workspace supports one team access per team. The [`tfe_team_access`](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/team_access) requires a team ID, so we can add [`tfe_team`](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/data-sources/team) data blocks to our plan for each team which allows reference to the team ID.

Usage

```yml
team_access: |-
  - team_name: Platform
    access: write
  - team_name: Backend
    permissions:
        runs: apply
        variables: write
        state_versions: read
        sentinel_mocks: none
        workspace_locking: true
  - team_name: ${data.terraform_remote_state.teams.outputs["read_only"].name}
    access: read
remote_state: |-
  teams:
    backend: remote
    config:
      hostname: app.terraform.io
      organization: ryanwholey
      workspaces:
        name: teams
```

### CODE REVIEW

**Comment by ryanwholey** [comment]
@bendrucker bump on this guy!

I have a PR branching off this one that adds tf validate tests for the workspace config. Caught some good bugs :)

File: action.yml [hunk 1]
```diff
@@ -78,6 +78,9 @@ inputs:
     required: false
   remote_states:
     description: Remote state blocks to configure in the workspace
+  team_access:
+    description: Teams and their associated permissions to be granted to the created workspaces
```
Comment (bendrucker) [inline action.yml:82]: Document the type of objects expected in these descriptions. And that they're YAML.

File: team_access_test.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package main
+
+import (
+	"testing"
+
+	"gotest.tools/v3/assert"
```
Comment (bendrucker) [inline team_access_test.go:6]: https://pkg.go.dev/github.com/stretchr/testify/assert is more active/popular

https://pkg.go.dev/github.com/google/go-cmp/cmp is good too

File: workspace.go [hunk 3]
```diff
@@ -208,3 +223,61 @@ func NewWorkspaceVariableResource(v *Variable) *WorkspaceVariableResource {
...
+	if len(teamAccess) == 0 {
+		return
+	}
+
+	ws.Data["tfe_team"] = map[string]interface{}{}
+	ws.Resources["tfe_team_access"] = map[string]interface{}{}
+
+	for _, ta := range teamAccess {
+
+		_, exists := ws.Data["tfe_team"][ta.TeamName]
```
Comment (bendrucker) [inline workspace.go:260]: Generally `ok` as the second bool argument, for map accesses, type assertions, etc

**Review by bendrucker — APPROVED** [review]

**Comment by ryanwholey** [comment]
Good comments, I'll follow up with them on the next PR 🙏

**Pull request merged**